### DISCUSSION TITLE
Cilium: Enabling `forwardKubeDNSToHost`and `bpf.masquerade: true` results in i/o timeout

### DISCUSSION THREAD

**Discussion by mrclrchtr** [siderolabs/talos#9157] [Q&A] [answered]
Follow up to: https://github.com/siderolabs/talos/issues/8836

I've been trying to set `bpf.masquerade: true` for a few hours. Unfortunately it does not work in any of the many configurations I have tested... `bpf.legacyHostRouting` makes no difference.

Just to be sure, does it work for you in the meantime/still in combination with `forwardKubeDNSToHost`?

**Comment by romario-07** [comment]
I encounter the same problem. I'm trying to use egress gateway which requires this settings:

`
   --set egressGateway.enabled=true
   --set bpf.masquerade=true
   --set kubeProxyReplacement=true`

**Reply by smira** [reply, marked as answer]
See https://github.com/siderolabs/talos/issues/10002#issuecomment-2573907377
